File: sample-custom-objects/src/main/java/com/quickblox/sample/customobjects/activities/BaseActivity.java
package com.quickblox.sample.customobjects.activities;

import android.app.ProgressDialog;
import android.os.Bundle;

import com.quickblox.sample.core.ui.activity.CoreBaseActivity;
import com.quickblox.sample.core.utils.DialogUtils;

import androidx.appcompat.app.ActionBar;

public abstract class BaseActivity extends CoreBaseActivity {

    protected ProgressDialog progressDialog;
    protected ActionBar actionBar;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        actionBar = getSupportActionBar();
        progressDialog = DialogUtils.getProgressDialog(this);
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
    }
}

File: sample-conference-kotlin/app/src/main/java/com/quickblox/sample/conference/kotlin/presentation/screens/base/BaseActivity.kt
package com.quickblox.sample.conference.kotlin.presentation.screens.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider

/*
 * Created by Injoit in 2021-09-30.
 * Copyright © 2021 Quickblox. All rights reserved.
 */
abstract class BaseActivity<VM : BaseViewModel>(private val viewModelClass: Class<VM>) : AppCompatActivity() {
    val viewModel by lazy {
        ViewModelProvider(this).get(viewModelClass)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        lifecycle.addObserver((viewModel as BaseViewModel).getViewStateObserver())
    }

    abstract fun showProgress()
    abstract fun hideProgress()
}